tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -2379,8 +2379,18 @@ tensorflow::Status ConvertSlice(OpConverterParams* params) {
         "Slice, at ",
         node_def.name());
   }
+  // Check that batch dimension is unmodified. If the batch size is -1 and the
+  // size is not -1, we do not convert the op since the batch dim could
+  // potentially be modified.
+  if ((size[0] != -1 && (input_dims[0] == -1 ||
```

Comment:
Please compare batch dim with 0 instead of -1. I think in some cases the batch dim is from graph_properties and it's possible that grappler output -2. All negative numbers means unknown, but grappler use different negative numbers to track symbolic dimensions.

Reply:
By compare batch dim with 0 do you mean `input_dims[0] < 0`?